packages/shell-api/src/interruptor.ts

```diff
@@ -75,12 +86,16 @@ export class InterruptFlag {
 
   /**
    * Mark an interrupt as having occurred.
+   *
+   * This should almost always be instantenous, although an additional listener
+   * installed through withOverrideInterruptBehavior() may perform additional
+   * cleanup work before the current connection is ready to be severed.
    */
-  public set(): void {
+  public async set(): Promise<void> {
     this.interrupted = true;
     const err = new MongoshInterruptedError();
     for (const listener of [...this.onInterruptListeners]) {
-      listener(err);
+      await listener(err);
```

Comment:
Do we want to catch any errors that the listener might throw so all listeners will always be notified?
```ts
try {
  await listener(err);
} catch {
  // ignore
}
```

Reply:
Yes, that seems like a good idea :+1: